documentation/examples/remote_storage/remote_storage_bridge/main.go

```diff
@@ -0,0 +1,232 @@
+	)
+	flag.StringVar(&cfg.listenAddr, "web.listen-address", ":9094", "Address to listen on for web endpoints.")
+	flag.StringVar(&cfg.telemetryPath, "web.telemetry-path", "/metrics", "Address to listen on for web endpoints.")
+
+	flag.Parse()
+
+	return cfg
+}
+
+func buildClients(cfg *config) []remote.StorageClient {
+	var clients []remote.StorageClient
+	if cfg.graphiteAddress != "" {
+		c := graphite.NewClient(
+			cfg.graphiteAddress, cfg.graphiteTransport,
+			cfg.remoteTimeout, cfg.graphitePrefix)
+		clients = append(clients, c)
+	}
+	if cfg.opentsdbURL != "" {
+		c := opentsdb.NewClient(cfg.opentsdbURL, cfg.remoteTimeout)
+		clients = append(clients, c)
+	}
+	if cfg.influxdbURL != "" {
+		url, err := url.Parse(cfg.influxdbURL)
+		if err != nil {
+			log.Fatalf("Failed to parse InfluxDB URL %q: %v", cfg.influxdbURL, err)
+		}
+		conf := influx.Config{
+			URL:      *url,
+			Username: cfg.influxdbUsername,
+			Password: cfg.influxdbPassword,
+			Timeout:  cfg.remoteTimeout,
+		}
+		c := influxdb.NewClient(conf, cfg.influxdbDatabase, cfg.influxdbRetentionPolicy)
+		prometheus.MustRegister(c)
+		clients = append(clients, c)
+	}
+	return clients
+}
+
+func serve(addr string, clients []remote.StorageClient) error {
+	http.HandleFunc("/receive", func(w http.ResponseWriter, r *http.Request) {
+		reqBuf, err := ioutil.ReadAll(snappy.NewReader(r.Body))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		var req remote.WriteRequest
+		if err := proto.Unmarshal(reqBuf, &req); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		samples := protoToSamples(&req)
+		receivedSamples.Add(float64(len(samples)))
+
+		var wg sync.WaitGroup
+		for _, c := range clients {
+			wg.Add(1)
+			go func(rc remote.StorageClient) {
+				sendSamples(rc, samples)
+				wg.Done()
+			}(c)
+		}
+		wg.Wait()
+	})
+
+	return http.ListenAndServe(addr, nil)
+}
+
+func protoToSamples(req *remote.WriteRequest) model.Samples {
+	var samples model.Samples
+	for _, ts := range req.Timeseries {
+		metric := make(model.Metric, len(ts.Labels))
+		for _, l := range ts.Labels {
+			metric[model.LabelName(l.Name)] = model.LabelValue(l.Value)
+		}
+
+		for _, s := range ts.Samples {
+			samples = append(samples, &model.Sample{
+				Metric:    metric,
+				Value:     model.SampleValue(s.Value),
+				Timestamp: model.Time(s.TimestampMs),
+			})
+		}
+	}
+	return samples
+}
+
+func sendSamples(c remote.StorageClient, samples model.Samples) {
+	begin := time.Now()
+	err := c.Store(samples)
+	duration := time.Since(begin).Seconds()
+	if err != nil {
+		log.Warnf("Error sending %d samples to remote storage %q: %v", len(samples), c.Name(), err)
+		failedSamples.WithLabelValues(c.Name()).Add(float64(len(samples)))
+	} else {
+		sentSamples.WithLabelValues(c.Name()).Add(float64(len(samples)))
+	}
+	sentBatchDuration.WithLabelValues(c.Name()).Observe(duration)
```

Comment:
This is fine.

The sentSamples should be a total though, so we have failures and total.